src/NumericalAlgorithms/Spectral/SwshCollocation.cpp

```diff
@@ -0,0 +1,70 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <cmath>
+
+#include "ErrorHandling/Assert.hpp"
+#include "NumericalAlgorithms/Spectral/ComplexDataView.hpp"
+#include "NumericalAlgorithms/Spectral/SwshCollocation.hpp"
+
+// IWYU pragma: no_include <sharp_geomhelpers.h>
+// IWYU pragma: no_include <sharp_lowlevel.h>
+
+namespace Spectral {
+namespace Swsh {
+
+template <ComplexRepresentation Representation>
+bool Collocation<Representation>::CollocationConstIterator::operator==(
```

Comment:
Would you mind adding a simple test like:
```cpp
CHECK(precomputed_collocation.begin() == precomputed_collocation.cbegin());
CHECK(precomputed_collocation.end() == precomputed_collocation.cend());
CHECK(precomputed_collocation.begin() != precomputed_collocation.end());
```
to test `operator==` and `begin`, `cbegin`, `end` and `cend`? (they're also implicitly tested by the range-based for loop :)

Reply:
Also, if you think you'll be iterating with this a lot, you should move the `operator==` into the header file class body so they get inlined :) That's pretty common for iterators since the way they bounds check is `!=`